recipes/libzip/all/conanfile.py

```diff
@@ -0,0 +1,86 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class LibZipConan(ConanFile):
+    name = "libzip"
+    description = "A C library for reading, creating, and modifying zip archives"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/nih-at/libzip"
+    license = "BSD-3-Clause"
+    topics = ("conan", "zip", "libzip", "zip-archives", "zip-editing")
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_lzma": [True, False],
+        "with_bzip2": [True, False],
+        "with_openssl": [True, False],
+        "enable_windows_crypto": [True, False],
```

Comment:
What is the effect of having both windows crypto and openssl enabled?

Reply:
In libzip CMakeLists.txt both openssl and windows crypto enabled by default:

```
option(ENABLE_OPENSSL "Enable use of OpenSSL" ON)
option(ENABLE_WINDOWS_CRYPTO "Enable use of Windows cryptography libraries" ON)
```

https://github.com/nih-at/libzip/blob/master/CMakeLists.txt

Reply:
It looks like the options are there to enable finding them, but not using them.
See https://github.com/nih-at/libzip/blob/d31cb3b2118914644ac4870297c1e9d294d1e8da/CMakeLists.txt#L207-L222
It looks like commoncrypto, windows crypto, gnutls, openssl and mbedtls are exclusive.
So I propose removing `with_openssl` and `enable_windows_crypto`,
and adding: `crypto: ["commoncrypto", "win32", "gnutls", "openssl", "mbedtls", "auto"]`.

By making the default "auto", you can use `win32` on Windows, and `openssl` on other platforms. In that case, you will also need to set `self.info.options.crypto` to the used option in `package_id`. See the cmake recipe for an example usage of `auto`.

Reply:
What is a correct behaviour if crypto == "win32" options is specified on non-Windows platform? Thank you.